lib/dockerregistry/inventory.go

```diff
@@ -1952,6 +1952,26 @@ func MKPopulateRequestsForPromotionEdges(
 	}
 }
 
+// RunChecks runs input PullChecks in order to check the promotion.
+func (sc *SyncContext) RunChecks(
+	edges map[PromotionEdge]interface{},
+	preChecks []PreCheck,
+) error {
+
+	var errors []error
+	for _, preCheck := range preChecks {
+		err := preCheck.Run(edges)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if errors != nil {
+		return fmt.Errorf("An error was encountered in one of the prechecks")
```

Comment:
This message is no longer accurate because you may have more than 1 error encountered.